src/components/app/ProjectPreview.svelte

```diff
@@ -85,9 +145,43 @@
                 : value
                   ? value.getRepresentativeText($locales)
                   : EXCEPTION_SYMBOL,
+            isCustomCharacter: false,
+            characterName: null,
         };
     });
 
+    // Add a state variable to hold the character
+    let character = $state<Character | null>(null);
+
+    // Effect to fetch the character from the database when characterName changes
+    $effect(() => {
+        // Clear character if characterName is not valid
+        if (!isCustomCharacter || !characterName) {
+            character = null;
+            return;
+        }
+
+        // Make sure DB.Characters exists and has getByName method
+        if (!DB.Characters || typeof DB.Characters.getByName !== 'function') {
```

Comment:
Why is this check for `getByName` here? It is always defined and always a function. The same is true of `DB.Characters`. Why is either check necessary?